internal/pkg/earpointer/analysis_test.go

```diff
@@ -651,3 +767,347 @@ func TestUnimplemented(t *testing.T) {
+	want := map[string]string{
+		"{f.d1[.],f.t1}":       "--> f.x",
+		"{f.d1}":               "[0->f.t1]",
+		"{f.s,f.t0,f.t2,f.t3}": "[0->f.t1]",
+		"{f.x}":                "[]",
+	}
+	wantstr := partitionItemsToString(want)
+	if got := state.String(); got != wantstr {
+		t.Errorf("got: %s\n want: %s", got, wantstr)
+	}
+}
+
+func TestVariadicCall(t *testing.T) {
+	code := `package p
+	func g(ks ...*int) *int {
+		return ks[0]
+	}
+	func f(a *int, b *int) *int {
+		return g(a, b)
+	}
+	`
+	/*
+		func f(a *int, b *int) *int:
+		0:                                                     entry P:0 S:0
+			t0 = new [2]*int (varargs)                         *[2]*int
+			t1 = &t0[0:int]                                    **int
+			*t1 = a
+			t2 = &t0[1:int]                                    **int
+			*t2 = b
+			t3 = slice t0[:]                                   []*int
+			t4 = g(t3...)                                      *int
+			return t4
+
+		func g(ks ...*int) *int:
+		0:                                                     entry P:0 S:0
+			t0 = &ks[0:int]                                    **int
+			t1 = *t0                                           *int
+			return t1
+	*/
+	state, err := runCode(code)
+	if err != nil {
+		t.Fatal(err)
+	}
+	want := map[string]string{
+		"{f.a,f.t4,g.t1}":  "[]",
+		"{f.b}":            "[]",
+		"{f.t0,f.t3,g.ks}": "[0->g.t0, 1->f.t2]",
+		"{f.t1,g.t0}":      "--> g.t1",
+		"{f.t2}":           "--> f.b",
+	}
+	wantstr := partitionItemsToString(want)
+	if got := state.String(); got != wantstr {
+		t.Errorf("got: %s\n want: %s", got, wantstr)
+	}
+}
+
+func TestMethodInvoke(t *testing.T) {
+	code := `package p
+	type T1 struct {}
+	func (T1) f(x *int) {}
+
+	type T2 struct {
+		T1
+	}
+	func g(t2 *T2, x *int) {
+		t2.f(x)
+	}
+	`
+	/*
+		func (arg0 T1) f(x *int):
+		0:                                          entry P:0 S:0
+			return
+
+		func (arg0 t.T2) f(x *int):
+		0:                                          entry P:0 S:0
+			t0 = local t.T2 ()                      *t.T2
+			*t0 = arg0
+			t1 = &t0.T1 [#0]                        *t.T1
+			t2 = *t1                                t.T1
+			t3 = (t.T1).f(t2, x)                    ()
+			return
+
+		func g(t2 *T2, x *int):
+		0:                                          entry P:0 S:0
+			t0 = &t2.T1 [#0]                        *T1
+			t1 = *t0                                T1
+			t2 = (T1).f(t1, x)                      ()
+			return
+	*/
+	state, err := runCode(code)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// T1.f.x and g.x are unified due to calling f().
+	// There are two separate f.arg0s w.r.t. T1.f and T2.f respectively.
+	want := map[string]string{
+		"{T1:f.arg0}":  "[]",
+		"{T1:f.x,g.x}": "[]",
+		"{T2:f.arg0}":  "[]",
+		"{T2:f.t0}":    "[]",
+		"{T2:f.t1}":    "[]",
+		"{T2:f.t2}":    "[]",
+		"{T2:f.x}":     "[]",
+		"{g.t0}":       "--> g.t1",
+		"{g.t1}":       "[]",
+		"{g.t2}":       "--> *g.t2",
```

Comment:
The name `t2` is a bit confusing here, because it's the name of both a param and a local temporary.